Client/Frontend/Browser/FaviconManager.swift

```diff
@@ -31,14 +36,81 @@ class FaviconManager : TabHelper {
     func scriptMessageHandlerName() -> String? {
         return "faviconsMessageHandler"
     }
+    
+    internal class FaviconError: MaybeErrorType {
```

Comment:
Nit: I would move this outside of this class scope. We don't really define our error types inside classes like this.